Makefile

```diff
@@ -114,521 +105,312 @@ endef
+$(MASPSX_APP): | $(VENV_DIR)
+	git submodule update --init $(dir $(MASPSX_APP))
+
+dependencies.pspeu: $(ALLEGREX) $(MWCCGAP_APP) $(MWCCPSP)
+$(MWCCGAP_APP): | $(VENV_DIR)
+	git submodule update --init $(dir $(MWCCGAP_APP))
+$(WIBO):
+	curl -sSfL -o $@ https://github.com/decompals/wibo/releases/download/0.6.13/wibo
+	$(muffle)sha256sum --check $(WIBO).sha256; chmod +x $(WIBO)
+$(MWCCPSP): $(WIBO) $(BIN_DIR)/mwccpsp_219
+
+dependencies.saturn: $(SATURN_SPLITTER_APP) $(ADPCM_EXTRACT_APP) $(DOSEMU_APP) $(CYGNUS)
+$(SATURN_SPLITTER_DIR)%:
+	git submodule update --init $(SATURN_SPLITTER_DIR)
+$(SATURN_SPLITTER_APP): $(SATURN_SPLITTER_DIR) $(SATURN_SPLITTER_DIR)/rust-dis/Cargo.toml $(wildcard $(SATURN_SPLITTER_DIR)/rust-dis/src/*)
+	cargo build --release --manifest-path $(SATURN_SPLITTER_DIR)/rust-dis/Cargo.toml
+$(ADPCM_EXTRACT_APP): $(SATURN_SPLITTER_DIR) $(SATURN_SPLITTER_DIR)/adpcm-extract/Cargo.toml $(wildcard $(SATURN_SPLITTER_DIR)/adpcm-extract/src/*)
+	cargo build --release --manifest-path $(SATURN_SPLITTER_DIR)/adpcm-extract/Cargo.toml
+$(DOSEMU_DIR):
+	cd tools; git clone https://github.com/sozud/dosemu-deb.git
+tools/dosemu.make.chkpt: $(DOSEMU_DIR)
+	sudo dpkg -i tools/dosemu-deb/*.deb && touch $@
+$(DOSEMU_APP): tools/dosemu.make.chkpt# This is just a pseudo target because the app binary is older than the repo so it always runs without a checkpoint
+
+$(ASMDIFFER_APP): | $(VENV_DIR)
+	git submodule update --init $(dir $(ASMDIFFER_APP))
+$(M2C_APP): tools/python-dependencies.make.chkpt | $(VENV_DIR)
+	git submodule update --init $(dir $(M2C_APP))
+$(PERMUTER_APP): | $(VENV_DIR)
+	git submodule update --init $(dir $(PERMUTER_APP))
+
+$(SOTNSTR_APP): tools/sotn_str/Cargo.toml $(wildcard tools/sotn_str/src/*)
+	cargo build --release --manifest-path tools/sotn_str/Cargo.toml
+$(SOTNDISK): $(GO) $(wildcard $(SOTNDISK_DIR)/*.go)
+	cd $(SOTNDISK_DIR) && $(GO) build
+$(SOTNASSETS): $(GO) $(wildcard $(SOTNASSETS_DIR)/*.go)
+	cd $(SOTNASSETS_DIR) && $(GO) build
+
+# Since venv is newly created, it can be reasonably assumed that the python requirements need to be installed
+$(VENV_DIR):
+	$(call echo,Creating python virtual environment) $(SYSTEM_PYTHON) -m venv $(VENV_DIR)
+	$(MAKE) python-dependencies
+tools/python-dependencies.make.chkpt: tools/requirements-python.txt | $(VENV_DIR)
+	$(PIP) install -r tools/requirements-python.txt && touch $@
+python-dependencies: tools/python-dependencies.make.chkpt
 
-define list_o_files_psp
-	$(foreach file,$(call list_src_files_psp,$(1)),$(BUILD_DIR)/$(file).o)
-endef
-### End old header ###
+tools/graphviz.make.chkpt: tools/python-dependencies.make.chkpt
+	sudo apt update && sudo apt-get install -y graphviz && touch $@
+graphviz: tools/graphviz.make.chkpt
 
-ifneq (,$(filter $(VERSION),us hd)) # Both us and hd versions use the PSX platform
+$(GO):
+	curl -sSfL -O https://go.dev/dl/go1.22.4.$(OS)-$(ARCH).tar.gz
+	$(muffle)tar -C $(HOME) -xzf go1.22.4.$(OS)-$(ARCH).tar.gz; rm go1.22.4.$(OS)-$(ARCH).tar.gz
```

Comment:
Doesn't really matter overall, but since I was already cleaning up the web download strings I added download of the correct go file for the system.